src/notification/services/email.service.ts

```diff
@@ -0,0 +1,117 @@
+import nodemailer from 'nodemailer';
+import { config } from '../../config';
+import { CurrencyRateEmailPayload, EmailPayload } from '../../common/models/email-payload';
+import logger from '../../common/services/logger.service';
+import { EventConsumer } from '../../common/services/messaging/event-consumer';
+import { SystemEvent, SystemEventType } from '../../common/models/system-event.model';
+import { RatesRepository } from '../data-access/repositories/rate.repository';
+import { Currency } from '../../rate/service/models/currency';
+import { SchedulerService } from '../../common/services/scheduler.service';
+import { SubscriptionsRepository } from '../../subscription/data-access/repositories/subscription.repository';
+import { SubscriptionsService } from '../../subscription/service/services/subscription.service';
+import { RateService } from './rate.service';
+
+export async function setupEmailService(messageConsumer: EventConsumer): Promise<EmailService> {
+  const emailService = new EmailService(messageConsumer);
+  await emailService.start();
+  return emailService;
+}
+
+export class EmailService {
+  private emailSender;
+
+  constructor(private eventConsumer: EventConsumer) {
+    this.emailSender = nodemailer.createTransport({
+      service: 'Gmail',
+      auth: {
+        user: config.api.emailServer.user,
+        pass: config.api.emailServer.password,
+      },
+    });
+    SchedulerService.initializeJob(config.cron.currencyRateEmailSchedule, async () => {
+      this.sendCurrencyRateEmail();
+    });
+  }
+
+  async start() {
+    await this.eventConsumer.addEventHandler(config.messageBroker.topics.email, (payload) => {
+      return this.systemEventHandler(payload);
+    });
+  }
+
+  public async sendCurrencyRateEmail(): Promise<void> {
+    try {
+      const rateRepository = new RateService(new RatesRepository());
```

Comment:
A new service instance is initialized each time the sendCurrencyRateEmail method is called. Embedding the rate service interface into EmailService might be a better approach, so you do not depend on a particular implementation.